src/main/kotlin/interpreter/stack/UserCodeStackFrame.kt

```diff
@@ -158,62 +158,51 @@ class UserCodeStackFrame(val function: PyUserFunction) : StackFrame() {
         // pool is the type we want to load
         val idx = opval.toInt()
         val loadResult = when (pool) {
-            LoadPool.CONST -> this.function.code.consts[idx]
+            LoadPool.CONST -> function.code.consts[idx]
             LoadPool.FAST -> {
-                val name = this.function.code.varnames[idx]
-                this.locals[name]!!
+                val name = function.code.varnames[idx]
+                locals[name]!!
             }
             LoadPool.NAME -> {
                 // sometimes a global...
-                val name = this.function.code.names[idx]
-                val realName = this.locals[name]
-
-                val result = if (realName == null) {
-                    val name = this.function.code.names[idx]
-                    this.function.getGlobal(name)
-                } else {
-                    realName
-                }
-                result
+                val name = function.code.names[idx]
+                locals[name] ?: function.getGlobal(name)
             }
             LoadPool.GLOBAL -> {
-                val name = this.function.code.names[idx]
-                this.function.getGlobal(name)
+                val name = function.code.names[idx]
+                function.getGlobal(name)
             }
             LoadPool.ATTR -> {
-                val toGetFrom = this.stack.pop()
-                val name = this.function.code.names[idx]
+                val toGetFrom = stack.pop()
+                val name = function.code.names[idx]
                 toGetFrom.pyGetAttribute(name)
             }
             LoadPool.METHOD -> {
                 // load_method sucks shit, for the record.
                 // we just treat this as an attribute load, for functions only
                 // because we already generated all the method wrappers anyway.
-                val toGetFrom = this.stack.pop()
-                val name = this.function.code.names[idx]
+                val toGetFrom = stack.pop()
+                val name = function.code.names[idx]
                 toGetFrom.pyGetAttribute(name)
             }
-            // interpreter error, not python error
-            else -> error("Unknown pool for LOAD_X instruction: $pool")
         }
 
-        this.stack.push(loadResult)
-        this.bytecodePointer += 1
+        stack.push(loadResult)
+        bytecodePointer += 1
     }
 
     /**
      * STORE_(NAME|FAST).
      */
     fun store(pool: LoadPool, arg: Byte) {
-        val idx = arg.toInt()
```

Comment:
Keep these temp variables, it makes more sense to have something called idx than just passing the argval

Reply:
Still needs resolving